scenario/build/generate.xml

```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0"?>
 <scenario xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:ece-tools:config/scenario.xsd">
+    <step name="apply-patches" type="Magento\MagentoCloud\Step\Build\ApplyPatches" priority="1"/>
```

Comment:
why priority 1? 
there won't chances to add something above

Reply:
This step must always be the first, before any manipulations.

Reply:
I do not agree, there still must be a possibility to add some step at the beginning of the scenario.

Reply:
@oshmyheliuk you can do this, but after patching :)

Reply:
No, you should change the priority

Reply:
There always be possibility to use `-100`, I would even set it to `0`

Reply:
So, what the point to set exactly 0 or 1 if there is still a possibility to set priority down to `-2147483648`? :)

Reply:
We set the lowest it makes sense for us, 3rd-party devs have this as a backdoor if they want something lower

Reply:
@shiftedreality , @BaDos , technically people may re-run composer or something similar before patching (in case our pre-built composer runs not in a way they want). So there are likely cases when this would need to be after something else. Can we confirm that negative values would actually work just to be sure we do not block this scenario?

Reply:
It seems like need to move this step
The log does not look logical
```
[datetime] INFO: Starting scenario(s): scenario/build/generate.xml (magento/ece-tools version: 2002.1.0, magento/magento2-base version: 2.3.3-p1)  
[datetime] NOTICE: Applying patches  
[datetime] NOTICE: End of applying patches
[datetime] INFO: Verbosity level is not set  
[datetime] INFO: Set Magento application mode to 'production'  
[datetime] NOTICE: Validating configuration 
```
I propose to change the priority to 150.
After `<step name="pre-build" type="Magento\MagentoCloud\Step\Build\PreBuild" priority="100"/>`

 ```
[datetime] INFO: Starting scenario(s): scenario/build/generate.xml (magento/ece-tools version: 2002.1.0, magento/magento2-base version: 2.3.3-p1)  
[datetime] INFO: Verbosity level is not set  
[datetime] NOTICE: Applying patches  
[datetime] NOTICE: End of applying patches
[datetime] INFO: Set Magento application mode to 'production'  
[datetime] NOTICE: Validating configuration 
```

Reply:
Yes, I think we can move `apply-patches` step after `pre-build` step, there is nothing related to Magento code in the `pre-build` step it shouldn't fail without applied patches.